### ISSUE TITLE
Resume Broken

### ISSUE THREAD

**Issue by kyleliang919** [facebookresearch/deit#173]
It keeps complaining about no state_dict_model_ema :
Failed to find state_dict_ema, starting from loaded model weights
But in the checkpoint there is clearly a model_ema:
dict_keys(['model', 'optimizer', 'lr_scheduler', 'epoch', 'model_ema', 'scaler', 'args'])

and the loss goes to NaN a few hundred steps after resume...

**Comment by TouvronHugo** [comment]
Hi @kyleliang919,
Thanks for your message.
Could you provide a little more context: Which model are you trying to train? with which command line?
Best,
Hugo

**Comment by kyleliang919** [comment]
yes, I was running deitIII with the pytorch distributed training since I didn't configure the slurm and run_with_submit, would that cause a major problem for the resume?
python -m torch.distributed.launch --nproc_per_node=4 --use_env --model deit_small_patch16_LS --data-path /path/to/imagenet --batch 256 --lr 4e-3 --epochs 800 --weight-decay 0.05 --sched cosine --input-size 224 --reprob 0.0 --node 1 --gpu 8 --smoothing 0.0 --warmup-epochs 5 --drop 0.0 --nb-classes 1000 --seed 0 --opt fusedlamb --warmup-lr 1e-6 --mixup .8 --drop-path 0.05 --cutmix 1.0 --unscale-lr --repeated-aug--bce-loss  --color-jitter 0.3 --ThreeAugment

**Comment by TouvronHugo** [comment]
Hi @kyleliang919,

No I don't think so.

We don't use the EMA in our experiments so you can use `--no-model-ema`. 
Did you notice any instabilities before resuming the checkpoint or did the code crash? 
Sometimes this can cause problems with the checkpoint.

What is a bit strange in your command  line you use the submitit flags: `--node` and `--gpu`. How many nodes do you use for training? The command  line is designed for 8 gpus, you have to readjust the hparams to use less gpus.

Best,

Hugo

**Comment by kyleliang919** [comment]
The code seems to be running fine without a crash, the loss is going to NaN after 400 steps after resume. 
I am running with 4 GPUs as indicated by the --nproc_per_node=4 
The default setting seems to be using ema and the example command doesn't seem to specify --no-model-ema, so I just leave it as it's.

**Comment by TouvronHugo** [comment]
Hi @kyleliang919,
It's strange, if there is a problem with the resume it should crash immediately and not after 400 steps.
In any case the command line is designed to be used with 8 GPUs (as indicated by `--nodes 1 --ngpus 8` ) with 4 GPUs the learning rate is too big which can cause instabilities. 
It is probably best to test with 8 GPUs to see if the problem persists.
Indeed, in the default setting there is the EMA but it has no impact on the training and is not used in the evaluation so it is not useful to keep it.
Best,
Hugo

**Comment by TouvronHugo** [comment]
Hi @Basavaraja-MS ,

Thanks for your message,

1. Yes, normally it's fixed now. 

2. Yes, we are aware of this issue (see [issue](https://github.com/facebookresearch/deit/issues/145#issue-1117914152)) it is necessary to modify the scheduler. We did not modify it for reproducibility reasons and because its impact on the training is quite limited. However, it's true that we can correct this by keeping an exception for the first 2 epochs. When I have more time (maybe in September) I can correct this. If you have time before then, don't hesitate to propose a PR.

**Comment by Basavaraja-MS** [comment]
Thanks @TouvronHugo  for quick feedback. 
-  From latest changes i can retrieve model ema successfully.
- Thanks for pointing to #145
